teacher.rb

```diff
@@ -0,0 +1,14 @@
+class Teacher < Person
+  attr_reader :id
+  attr_accessor :specialization
+
+  def initialize(specialization)
+    super()
```

Comment:
- Kindly add the `name`, and `age` here also to trigger a call to the parent constructor.